slither/solc_parsing/declarations/using_for_top_level.py

```diff
@@ -101,6 +108,14 @@ def _analyze_top_level_function(
                 self._propagate_global(type_name)
                 break
 
+    def _analyze_operator(
+        self, operator: str, function_name: str, type_name: TypeAliasTopLevel
+    ) -> None:
+        for tl_function in self.compilation_unit.functions_top_level:
+            if tl_function.name == function_name:
```

Comment:
This should check that the parameters are the correct type otherwise I think it will run into bugs similar to https://github.com/crytic/slither/pull/1625